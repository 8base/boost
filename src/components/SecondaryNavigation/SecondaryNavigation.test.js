// @flow
import React from 'react';
import { SecondaryNavigation } from './';

describe('<SecondaryNavigation />', () => {
  it('should shallow SecondaryNavigation', () => {
    const wrapper = shallow(
      <SecondaryNavigation>
        <SecondaryNavigation.Item label="First item" />
      </SecondaryNavigation>,
    );

    expect(wrapper).toMatchSnapshot();
  });


  it('should shallow SecondaryNavigation item', () => {
    const wrapper = shallow(
      <SecondaryNavigation.Item label="First item" />,
    );

    expect(wrapper).toMatchSnapshot();
  });


  it('should render SecondaryNavigation components content', () => {
    const wrapper = mount(
      <EightBaseBoostProvider>
        <SecondaryNavigation>
          <SecondaryNavigation.Item label="First item" />
          <SecondaryNavigation.Item label="Second item" className="active" />
          <SecondaryNavigation.Item label="Third item" />
          <SecondaryNavigation.Item label="Fourth item" />
        </SecondaryNavigation>
      </EightBaseBoostProvider>,
    );

    expect(wrapper.find(SecondaryNavigation.Item).at(0).text()).toBe('First item');
    expect(wrapper.find(SecondaryNavigation.Item).at(3).text()).toBe('Fourth item');
  });
});

