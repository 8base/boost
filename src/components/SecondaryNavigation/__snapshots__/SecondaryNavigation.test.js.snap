// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SecondaryNavigation /> should shallow SecondaryNavigation 1`] = `
<Boost(secondaryNavigation)
  tagName="nav"
>
  <SecondaryNavigationItem
    label="First item"
  />
</Boost(secondaryNavigation)>
`;

exports[`<SecondaryNavigation /> should shallow SecondaryNavigation item 1`] = `
<Boost(secondaryNavigationItem)
  hovered={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  tagName="div"
>
  <Boost(secondaryNavigationItemLabel)
    modifiers={Object {}}
    tagName="span"
  >
    First item
  </Boost(secondaryNavigationItemLabel)>
</Boost(secondaryNavigationItem)>
`;
